src/common/utils.cc

```diff
@@ -41,7 +41,9 @@ NormalizeVec(DataType* x, int32_t d) {
     } else if constexpr (std::is_same_v<DataType, bf16>) {
         norm_l2_sqr = faiss::bf16_vec_norm_L2sqr(x, d);
     } else {
-        LOG_KNOWHERE_ERROR_ << "unknown datatype";
+        for (int32_t i = 0; i < d; i++) {
+            norm_l2_sqr += (float)x[i] * (float)x[i];
+        }
```

Comment:
why we don't throw exception here? the float cast is an undefined operation here.

Reply:
updated